my_each.rb

```diff
@@ -1,6 +1,143 @@
 module Enumerable
-    def my_each(array)
-      for i in 0...array.length
-        p "gaa"
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
     end
-end
\ No newline at end of file
+    self
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield self[i], i
+    end
+    self
+  end
+
+  def my_select
+    new_array = []
+    my_each do |element|
+      new_array.push(element) if yield element
+    end
+    new_array
+  end
+
+  def my_all?
+    result = true
+    my_each do |element|
+      statement = if block_given?
+                    yield element
+                  else
+                    element
+                  end
+
+      unless statement
+        result = false
+        break
+      end
+
+      result = true
+    end
+    result
+  end
+
+  def my_any?
+    result = true
+    my_each do |element|
+      statement = if block_given?
+                    yield element
+                  else
+                    element
+                  end
+
+      if statement
+        result = true
+        break
+      end
+      result = false
+    end
+    result
+  end
+
+  def my_none?
+    !my_any? do |element|
+      yield element
+    end
```

Comment:
Please don't rely on the other method to create this one. The purpose is to think of the logic. So you know it's the opposite of the other, then write it.